token-client/src/main/java/com/sap/cloud/security/xsuaa/tokenflows/CacheConfiguration.java

```diff
@@ -0,0 +1,63 @@
+package com.sap.cloud.security.xsuaa.tokenflows;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * Data class to capture configuration options of caches.
+ */
+public class CacheConfiguration {
+
+	/**
+	 * Creates the default cache configuration.
+	 */
+	public static final CacheConfiguration DEFAULT = new CacheConfiguration(Duration.ofMinutes(15), 100);
+
+	/**
+	 * Special cache configuration used to disable caching.
+	 */
+	public static final CacheConfiguration CACHE_DISABLED = new CacheConfiguration(Duration.ZERO, 0);
+
+	private final Duration expireAfterWrite;
+	private final int cacheSize;
+
+	public CacheConfiguration(Duration expireAfterWrite, int cacheSize) {
```

Comment:
i would prefer a private Constructor and a getInstance() method

Reply:
Done :)